crates/ruff/src/rules/flake8_simplify/rules/ast_if.rs

```diff
@@ -988,3 +988,82 @@ pub(crate) fn use_dict_get_with_default(checker: &mut Checker, stmt_if: &ast::St
     }
     checker.diagnostics.push(diagnostic);
 }
+
+/// SIM401
+// var = a_dict[key] if key in a_dict else "default3"
+pub(crate) fn if_expr_with_dict(
+    checker: &mut Checker,
+    expr: &Expr,
+    test: &Expr,
+    body: &Expr,
+    orelse: &Expr,
+) {
+    let Expr::Compare(ast::ExprCompare {
+        left: test_key,
+        ops,
+        comparators: test_dict,
+        range: _,
+    }) = test
+    else {
+        return;
+    };
+    let [test_dict] = test_dict.as_slice() else {
+        return;
+    };
+
+    if let [CmpOp::NotIn] = ops[..] {
+        return;
+    }
```

Comment:
What's the motivation behind not adding support for the `not in` comparison?

```python
var = "default" if key not in data else data[key]
# The above code could be converted to `var = data.get(key, "default")`
```
I think it is supported in the expanded version, right? This is what I get:

```diff
-if key not in data:
-    var = "default"
-else:
-    var = data[key]
+var = data.get(key, "default")
```

Reply:
I do not supported `not in` comparison because it is not valid python code
```python
var = "default" if key not in data else data[key]
Traceback (most recent call last):
  File "<string>", line 1, in <module>
NameError: name 'key' is not defined
```

Reply:
That error is because you have not defined `key` not because it's invalid syntax

```python
key = "test"
data = {}

var = "default" if key not in data else data[key]
var = data[key] if key in data else "default"

if key not in data:
    var = "default"
else:
    var = data[key]


if key in data:
    var = data["key"]
else:
    var = "default"

data.get("key", "default")
```